app/models/concerns/orchestration/compute.rb

```diff
@@ -124,16 +128,21 @@ def setComputeDetails
       attrs.each do |foreman_attr, fog_attr|
         if foreman_attr == :mac
           return false unless match_macs_to_nics(fog_attr)
-        elsif foreman_attr == :ip
-          value = vm.send(fog_attr) || find_address
+        elsif [:ip, :ip6].include?(foreman_attr)
+          value = vm.send(fog_attr) || find_address(foreman_attr)
           self.send("#{foreman_attr}=", value)
           return false unless validate_foreman_attr(value, ::Nic::Base, foreman_attr)
```

Comment:
If the provider claims to provide :ip6 but it doesn't (e.g. foreman_digitalocean, don't tick IPv6), then this still throws an error if `value.blank?` and there's another blank value in the database. Don't call it if the value's blank?

    2016-11-01T14:36:30 c50b235a [sql] [W] ip6  is already used by